nin/backend/layer.js

```diff
@@ -0,0 +1,95 @@
+var fs = require('fs');
+
+function add(projectPath, layer, callback) {
+  read(projectPath, function (err, layers) {
+    if (err) {
+      if (callback) {
+        callback(err);
+      }
+      return;
+    }
+    var defaultLayer = {
+      type: '',
+      displayName: '',
+      startFrame: 0,
+      endFrame: 1000,
+      color: 'red',
+      config: {}
+    };
+    layers.unshift(mergeOptions(layer, defaultLayer));
+    write(projectPath, layers, function (err) {
+      if (callback) {
+        callback(err);
+      }
+    });
+  });
+}
+
+function update(projectPath, layerId, layer, callback) {
+  read(projectPath, function (err, layers) {
+    if (err) {
+      if (callback) {
+        callback(err);
+      }
+      return;
+    }
+    for (var key in layer) {
+      layers[layerId][key] = layer[key];
+    }
+    write(projectPath, layers, function (err) {
+      if (callback) {
+        callback(err);
+      }
+    });
+  });
+}
+
+var operationQueue = [];
+var active = null;
+
+function read(projectPath, callback) {
+  operationQueue.push([projectPath, callback]);
+  if (active === null) {
+    advanceQueue();
+  }
+}
+
+function write(projectPath, layers, callback) {
+  var data = JSON.stringify(layers, null, 2) + '\n';
+  fs.writeFile(projectPath + '/res/layers.json', data, function(err) {
+    advanceQueue();
+    callback(err);
+  });
+}
+
+function advanceQueue() {
+  active = operationQueue.shift() || null;
```

Comment:
ah it's set further up. I thought it was only used for the check under, and undefined is falsy.
Wait, shouldn't falsy be equally good as null then?